Force Homebrew update to work around CircleCI auto-update issue

chriseth opened the pull request.

cameel committed:
osx_install_dependencies.sh: Use set -ue to fail fast on errors

cameel commented:
For a moment I thought I was not getting any output from `brew`. Finally I realized that it just somehow breaks the output updater on CircleCI and the whole output only shows after refreshing the page.

So, anyway, the problem is that `brew install boost` fails because it cannot download `icu4c`:

```
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/3.1.2\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.14.4\)\ curl/7.54.0 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://homebrew.bintray.com/icu4c-68.2.mojave.bottle.tar.gz
==> Downloading https://homebrew.bintray.com/icu4c-68.2.mojave.bottle.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/3.1.2\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.14.4\)\ curl/7.54.0 --header Accept-Language:\ en --retry 3 --location --range 0-1 --dump-header - --write-out \%\{http_code\} --output /dev/null https://homebrew.bintray.com/icu4c-68.2.mojave.bottle.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/3.1.2\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.14.4\)\ curl/7.54.0 --header Accept-Language:\ en --fail --retry 3 --location --remote-time --continue-at 0 --output /Users/distiller/Library/Caches/Homebrew/downloads/7117aecde2df87533a7a66f5246ff4137df4dd51e660b62edf1d45a73dd7cb51--icu4c-68.2.mojave.bottle.tar.gz.incomplete https://homebrew.bintray.com/icu4c-68.2.mojave.bottle.tar.gz --connect-timeout 15
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 Not Found
Trying a mirror...
==> Downloading https://ghcr.io/v2/homebrew/core/icu4c-68.2.mojave.bottle.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/3.1.2\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.14.4\)\ curl/7.54.0 --header Accept-Language:\ en --retry 3 --location --silent --head --request GET https://ghcr.io/v2/homebrew/core/icu4c-68.2.mojave.bottle.tar.gz
==> Downloading from https://github.com/-/v2/packages/container/package/homebrew%2Fcore%2Ficu4c-68.2.mojave.bottle.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/3.1.2\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.14.4\)\ curl/7.54.0 --header Accept-Language:\ en --retry 3 --location --range 0-1 --dump-header - --write-out \%\{http_code\} --output /dev/null https://github.com/-/v2/packages/container/package/homebrew\%2Fcore\%2Ficu4c-68.2.mojave.bottle.tar.gz
/usr/bin/curl --disable --globoff --show-error --user-agent Homebrew/3.1.2\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.14.4\)\ curl/7.54.0 --header Accept-Language:\ en --fail --retry 3 --location --remote-time --continue-at 0 --output /Users/distiller/Library/Caches/Homebrew/downloads/7117aecde2df87533a7a66f5246ff4137df4dd51e660b62edf1d45a73dd7cb51--icu4c-68.2.mojave.bottle.tar.gz.incomplete https://github.com/-/v2/packages/container/package/homebrew\%2Fcore\%2Ficu4c-68.2.mojave.bottle.tar.gz --connect-timeout 15
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404 
Error: Failed to download resource "icu4c"
Download failed: https://ghcr.io/v2/homebrew/core/icu4c-68.2.mojave.bottle.tar.gz
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:407:in `rescue in fetch'
/usr/local/Homebrew/Library/Homebrew/download_strategy.rb:404:in `fetch'
/usr/local/Homebrew/Library/Homebrew/resource.rb:142:in `fetch'
/usr/local/Homebrew/Library/Homebrew/software_spec.rb:334:in `fetch'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:1120:in `fetch'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:687:in `fetch_dependency'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:1105:in `block in fetch_dependencies'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:1105:in `each'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:1105:in `fetch_dependencies'
/usr/local/Homebrew/Library/Homebrew/formula_installer.rb:1110:in `fetch'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:421:in `install_formula'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:324:in `block in install'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:322:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/install.rb:322:in `install'
/usr/local/Homebrew/Library/Homebrew/brew.rb:122:in `<main>'

Exited with code exit status 1
```

These URLs do give me HTTP 404 when I visit them in the browser so the file is actually missing.
And it's not that it's outdated. 68.2 is the latest version of [icu4c](https://formulae.brew.sh/formula/icu4c).

cameel commented:
This seems to be our problem: https://github.com/Homebrew/brew/issues/11123#issuecomment-818393020
- Homebrew has recently (last 1-2 months?) updated their bintray URLs (https://github.com/Homebrew/brew/issues/10739). So we need a recent version for it to work.
- The version in CircleCI is very old but it runs an update automatically when we install `boost`. For some reason it is not enough though. Apparently this is an issue on CircleCI side. An explicit `brew update` is a workaround.

cameel committed:
osx_install_dependencies.sh: Update brew explicitly to workaround auto-update issue on CircleCI

- Old versions have problems with binaries hosted on bintray. We need an up-to-date one.
- It automatically updates itself when we install boost but apparently there's a problem with that on CircleCI. An explicit update works.

cameel commented:
I think this fixes the problem now. I have squashed the commits and if it passes all tests it should be good to merge.

cameel commented:
Tests are failing. But these are just external contract tests I think it's due to some transient npm problem. They're timing out during package installation:

```
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...

Too long with no output (exceeded 10m0s): context deadline exceeded
```

So rerunning them might be enough to make them pass.
